sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1404,7 +1404,10 @@ case class ParseUrl(children: Seq[Expression])
     try {
       new URI(url.toString)
     } catch {
-      case e: URISyntaxException => null
+      // We fail on error if in ansi mode.
+      case e: URISyntaxException if SQLConf.get.ansiEnabled =>
+        throw new IllegalArgumentException(s"Find an invaild url string ${url.toString}", e)
```

Comment:
To clarify, the raw exception message is something like `Illegal character in query at index 33: https://a.b.c/index.php?params1=a|b&params2=x`.